src/buildout/wheel/__init__.py

```diff
@@ -1,56 +1,230 @@
+       just the eggs that `Environment` would auto-detect
+
+     * knows how to install `wheels`.
+
+     * treats all dists inside `eggs-directory` as eggs, even if they're, say,
+       unpacked distros from wheels.
+    """
+
+    def _get_full_path(self):
+        """ Prepend all dists found in self._dest to self._path """
+        # TODO: This method should go into zc.buildout proper.
+        return self._get_dest_dist_paths() + self._path
+
+    def _get_dest_dist_paths(self):
+        # TODO: This method should go into zc.buildout proper.
+        dest = self._dest
+        if dest is None:
+            return []
+        eggs = glob.glob(os.path.join(dest, '*.egg'))
+        dists = [os.path.dirname(dist_info) for dist_info in
+                 glob.glob(os.path.join(dest, '*', '*.dist-info'))]
+        # sort them like pkg_resources.find_on_path() would
+        return pkg_resources._by_version_descending(set(eggs + dists))
+
+    def _env_rescan_dest(self):
+        # TODO: This method should go into zc.buildout proper.
+        self._env.scan(self._get_dest_dist_paths())
+
+    def _maybe_eggify(self, dist):
+        """ Make sure everything installed under `._dest` is seen as an egg """
+        # TODO: This method and its two uses below can go in buildout proper
+        # without altering its current behavior.
+        if dist.location.startswith(self._dest + '/'):
+            dist.precedence=pkg_resources.EGG_DIST
+        return dist
+
+    def _env_get_best_match(self, requirement, ws):
+        dist = orig_Installer._env_get_best_match(self, requirement, ws)
+        return self._maybe_eggify(dist)
+
+    def _env_get_dists_for(self, project_name):
+        dists = orig_Installer._env_get_dists_for(self, project_name)
+        return [self._maybe_eggify(dist) for dist in dists]
+
+    def _call_easy_install(self, spec, ws, dest, dist):
+        if not spec.endswith('.whl'):
+            return orig_Installer._call_easy_install(spec, ws, dest, dist)
+        tmp = tempfile.mkdtemp(dir=dest)
+        try:
+            # this causes the dist tree to be written twice in tempdirs inside
+            # `dest`, but the alternative would be to trick
+            # `setuptools.archive_util.unpack_archive` into unpacking wheels
+            # which requirs dealing with
+            # `setuptools.archive_util.unpack_archive.default_filter`.
+            dist = WheelInstaller(spec).install_into(tmp)
+            move = zc.buildout.easy_install._move_to_eggs_dir_and_compile
+            newloc = move(dist, dest)
+            return pkg_resources.Environment([newloc])[dist.project_name]
+        finally:
+            shutil.rmtree(tmp)
 
-wheel_supported = pip.pep425tags.get_supported
 
-def interpret_distro_name(
-    location, basename, metadata, py_version=None,
-    precedence=pkg_resources.SOURCE_DIST,
-    platform=None
-    ):
-    if '.whl' in location:
-        path = parse.urlparse(location).path
+class WheelInstaller(object):
```

Comment:
There's really nothing magical about it now. To understand what it does I recommend just searching for it's instantiations and following its uses.